python/lsst/dax/imgserv/imageREST_v1.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#; GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This module implements the RESTful interface for Image Cutout Service.
+Corresponding URI: /image
+
+@author: John Gates, SLAC
+@author: Brian Van Klaveren, SLAC
+@author: Kenny Lo, SLAC
+
+"""
+import os
+import tempfile
+import traceback
+import json
+
+from flask import Blueprint, make_response, request, current_app, jsonify
+from flask import render_template
+
+import lsst.log as log
+
+from http.client import BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND
+from .locateImage import (
+        image_open_v1, W13DeepCoaddDb, W13RawDb, W13CalexpDb
+)
+
+from .image_v1 import Image
+from .dispatch_v1 import Dispatcher
+from .hashutil import Hasher
+
+imageRESTv1 = Blueprint("imageRESTv1", __name__, static_folder="static",
+                      template_folder="templates")
+
+# To be called from webserv
+def load_imgserv_config(config_path, db_auth_conf):
+    """Load configuration info into ImageServ."""
+    if config_path is None:
+        # use default root_path for imageRESTv1
+        config_path = imageRESTv1.root_path+"/config/"
+    f_json = os.path.join(config_path, "settings.json")
+    # load the general config file  
+    current_app.config.from_json(f_json)
+    # configure the log file (log4cxx)
+    log.configure(os.path.join(config_path, "log.properties"))
+    current_app.config["DAX_IMG_DBCONF"] = db_auth_conf
+    current_app.config["DAX_IMG_CONFIG"] = config_path
+
+@imageRESTv1.route("/")
+def index():
+    return make_response(render_template("index_v1.html"))
+
+@imageRESTv1.route("/availability", methods=["GET"])
+def getimage_avail():
+    return _getimage_avail(request)
+
+@imageRESTv1.route("/capabilities", methods=["GET"])
+def getimage_capabilities():
+    return _getimage_capabilities(request)
+
+@imageRESTv1.route("/<db_id>", methods=["GET"])
+def getimage_sync(db_id):
+    return _getimage(request, db_id)
+
+@imageRESTv1.route("/<db_id>/async", methods=["POST"])
+def getimage_async(db_id):
+    return _getimage_async(request)
```

Comment:
I don't see _getimage_async implemented anywhere yet.

`raise NotImplementedError("async endpoint not yet implemented")`

Reply:
Fixed as advised here.